scripts/setup-rg/rg-template.json

```diff
@@ -0,0 +1,376 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "projectPrefix": {
+            "type": "string",
+            "metadata": {
+                "description": "Prefix used for naming resources."
+            },
+            "defaultValue": "mlos"
+        },
+        "vmSKU": {
+            "type": "string",
+            "defaultValue": "Standard_D2s_v3",
+            "metadata": {
+                "description": "VM SKU"
+            }
+        },
+        "vmAdminUsername": {
+            "type": "string"
+        },
+        "sshPublicKey": {
+            "type": "string"
+        },
+        "fileShareName": {
+            "type": "string",
+            "defaultValue": "mlos-file-share"
+        },
+        "resultsDbName": {
+            "type": "string"
+        },
+        "resultsDbAdminUsername": {
+            "type": "string",
+            "defaultValue": "mlos"
+        },
+        "resultsDbAdminPassword": {
+            "type": "securestring"
+        }
+    },
+    "functions": [],
+    "variables": {
+        "storageAccountName": "[toLower(concat(replace(parameters('projectPrefix'), '-', ''), 'storage'))]",
+        "vmName": "[concat(parameters('projectPrefix'), '-vm')]",
+        "networkInterfaceName": "[concat(variables('vmName'), '-ni-', uniqueString(resourceGroup().id))]",
```

Comment:
why append the RG to the NIC name?

Reply:
It seemed to be a pattern from the ARM documentation, wasnt sure if NI names needed to be unique this way. I realize in other templates we have this was not needed, I will look into removing this!